cmd/cli/app/provider/provider_enroll.go

```diff
@@ -275,8 +283,10 @@ func init() {
 	ProviderCmd.AddCommand(enrollCmd)
 	// Flags
 	enrollCmd.Flags().StringP("token", "t", "", "Personal Access Token (PAT) to use for enrollment")
```

Comment:
The token is also for legacy GitHub only

Reply:
for now. Other providers (e.g. DockerHub) will make use of this.

Reply:
The problem is that with the current approach, this command will end up with every possible optional flag for every possible provider. In the Minder CLI proposal document, I have suggested that we move to a structure with subcommands per provider, e.g.:

```
minder provider enroll github-app [flags]
minder provider enroll github [flags]
minder provider enroll oci [flags]
```

However, since we need to maintain backwards compatibility, I am not making this change at this point.

Ria's feedback does raise the issue of whether we should avoid calling certain parts of the code if we are creating a `github-app` provider

Reply:
Eventually, it would be nice if we could use the `AuthorizationFlows` for each provider to create a selection box for the user. For example the legacy `github` provider support the OAuth flow and the user input flow, so when they run `minder provider enroll --class github` we could show:
```
Select your authorization flow:
[] provider the token myself
[] use OAuth 2.0 authorization code flow to generate a token 
```

Then we remove `--token` from the parameters of `enroll` and new providers just have to define their accepted authorization flows.